### ISSUE TITLE
Processing model running much slower in the LTR version

### ISSUE THREAD

**Issue by leonazareth** [qgis/QGIS#58224]
### What is the bug or the crash?

I have a processing model that runs in 0.55 seconds in version 3.36.3-Maidenhead, but running in the current LTR version (3.34.9-Prizren) takes 231.15 seconds (3 minutes 51 seconds) to run.

I tested it on two different computers with Win11 and Win10, respectively, and the results were similar.

Following are the logs of running in the two different versions of QGIS:
[ProcessingModelLog_QGIS3.3.34.9_MSI.txt](https://github.com/user-attachments/files/16353666/ProcessingModelLog_QGIS3.3.34.9_MSI.txt)
[ProcessingModelLogQGIS3.36.3_MSI.txt](https://github.com/user-attachments/files/16353667/ProcessingModelLogQGIS3.36.3_MSI.txt)

I am also sending the example project and the model that I used on the test:
[QGISProcessingModelerProj.zip](https://github.com/user-attachments/files/16353700/QGISProcessingModelerProj.zip)

### Steps to reproduce the issue

1. Unzip the files on the .zip file
2. Open the folder
3. Open the project using the LTR version of QGIS
4. Go to the option of opening an existing model on QGIS and select the file "01 subareas_delimitation.model3"
5. Configure the processing as the image bellow:
![ModelProcessingView](https://github.com/user-attachments/assets/071b8580-4cdb-437f-9756-bca272d351ac)

### Versions

3.34.9-Prizren

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

Summing the time of the 42 steps individually, the result is 0.178 seconds. It seems that the problem is between the execution of one process and the next one.

**Comment by dingausmwald** [comment]
Yes. Just had the same conclusion after updating to 38. Just that my script ran 50 min. Now it finishes in 2. It says 0.003 sec to execute, but waits minutes between the steps.

**Comment by arnoPujol** [comment]
Hello,

I'm running QGIS version 3.34.7-Prizren and I could observe the same thing.
Each individual process of my model is relatively fast, summing up to 48.787s however it took 30 minutes and 16 seconds to run the whole model.
